src/app/content/components/NudgeStudyTools/index.tsx

```diff
@@ -121,6 +128,44 @@ const NudgeStudyTools = () => {
   </NudgeWrapper>;
 };
 
+function CloseButtonHoldingFocus({
+  positions,
+  dismiss,
+}: {
+  positions: Exclude<ReturnType<typeof usePositions>, null>;
+  dismiss: () => void;
+}) {
+  const ref = React.useRef<HTMLElement>();
+
+  React.useEffect(
```

Comment:
Suggestion: this seems like a helpful hook, it might be nice if this lived somewhere in utils that we could import in other overlays.

Reply:
I didn't find another file where I thought it belonged, but I did separate it out as a hook within the file.